addons/notes/src/register.tsx

```diff
@@ -1,18 +1,44 @@
-import React from 'react';
-import PropTypes from 'prop-types';
+import * as React from 'react';
 import addons from '@storybook/addons';
 
 import styled from '@emotion/styled';
 
+interface NotesChannel {
+  on: (listener: string, callback: (text: string) => void) => any; // todo check correct return value definition
+  emit: any; // todo check correct definition
+  removeListener: (listener: string, callback: (text: string) => void) => void;
+}
+
+interface NotesApi {
+  onStory: (callback: () => void) => void;
+  getQueryParam: any; // todo check correct definition
+  setQueryParams: any; // todo check correct definition
+}
+
+interface NotesProps {
+  active: boolean;
+  channel: NotesChannel;
+  api: NotesApi;
+}
+
+interface NotesState {
+  text: string;
+}
+
 const Panel = styled.div({
   padding: 10,
   boxSizing: 'border-box',
   width: '100%',
 });
 
-export class Notes extends React.Component {
-  constructor(...args) {
-    super(...args);
+export class Notes extends React.Component<NotesProps, NotesState> {
+
+  // todo check how react typescript does this, especially unmounted
+  stopListeningOnStory: any; // todo check correct definition
+  unmounted: any; // todo check correct definition
+
+  constructor(...args: any) {
+    super(args /* todo: This was ...args before, check if this was a bug */);
```

Comment:
Typescript is _smart_, so:
```
super(args);      // will be transformed to _super.call(this, args)
super(...args);   // will be transformed to _super.apply(this, args)
```
So your change will cause a bug. If you want to use spread - write 
```
constructor(...args: any[])
```
Anyway - there is no need to define a constructor with React, see my comment bellow